src/display/font_loader.js

```diff
@@ -71,13 +79,54 @@ class FontLoader {
     }
   }
 
+  async loadSystemFont(info) {
+    assert(
+      !this.disableFontFace,
+      "loadSystemFont shouldn't be called when `disableFontFace` is set."
+    );
+
+    const { loadedName, src, style } = info;
+    if (this.#systemFonts.has(src)) {
+      return;
+    }
+    this.#systemFonts.add(src);
+    if (this.isFontLoadingAPISupported) {
+      const fontFace = new FontFace(loadedName, src, style);
+      this.addNativeFontFace(fontFace);
+      try {
+        await fontFace.load();
+      } catch {
+        warn(
+          `Cannot load system font: ${loadedName} for style ${style.style} and weight ${style.weight}.`
+        );
+        this.removeNativeFontFace(fontFace);
+      }
+      return;
+    }
+
+    if (this._document) {
+      const rule =
+        "@font-face { " +
+        `font-family: ${loadedName};` +
+        `src: ${src};` +
+        `font-style: ${style.style};` +
+        `font-weight: ${style.weight}; }`;
+      this.insertRule(rule);
```

Comment:
Another question, don't we need to add code [like this one](https://github.com/mozilla/pdf.js/blob/797f8d3dca5e0770c83788b3b65387884031caa0/src/display/font_loader.js#L103-L113) here as well to ensure that we actually wait for the font to be loaded in this case too?

Reply:
I added the code to wait for the font loading... but it doesn't work.
I suppose that the font we create to check its availability is created before the one loaded from `src`.
What's the context of not having font loading api ?